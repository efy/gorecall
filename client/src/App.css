.app {
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.app-logo {
  width: 100px;
  height: 100px;
}
