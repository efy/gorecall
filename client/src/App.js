import React, { Component } from 'react'
import { Redirect, Switch, Route, Link } from 'react-router-dom'
import 'spectre.css'
import './App.css'

import Sidebar from './common/Sidebar'

import BookmarksPage from './pages/Bookmarks'
import BookmarkPage from './pages/Bookmark'
import NewBookmarkPage from './pages/NewBookmark'
import TagsPage from './pages/Tags'
import TagPage from './pages/Tag'
import NewTagPage from './pages/NewTag'
import SettingsPage from './pages/Settings'
import NotFoundPage from './pages/NotFound'

class App extends Component {
  render() {
    return (
      <div className="app">
        <div className="app-left">
          <Sidebar />
        </div>
        <div className="app-right">
          <main className="app-main">
            <Switch>
              <Route exact path="/">
                <Redirect to="/bookmarks" />
              </Route>

              <Route exact path="/bookmarks">
                <BookmarksPage />
              </Route>

              <Route exact path="/bookmarks/search">
                <BookmarksPage />
              </Route>

              <Route exact path="/bookmarks/new">
                <NewBookmarkPage />
              </Route>

              <Route exact path="/bookmarks/:id">
                <BookmarkPage />
              </Route>

              <Route exact path="/tags">
                <TagsPage />
              </Route>

              <Route exact path="/tags/new">
                <NewTagPage />
              </Route>

              <Route exact path="/tags/:id">
                <TagPage />
              </Route>

              <Route path="/settings">
                <SettingsPage />
              </Route>

              <Route>
                <NotFoundPage />
              </Route>
            </Switch>
          </main>
        </div>
      </div>
    );
  }
}

export default App;
