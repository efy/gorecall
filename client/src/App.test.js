import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import App from './App';

describe('App', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.exists()).toBe(true)
  });

  it('renders the logo', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.exists()).toBe(true)
  })

  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
