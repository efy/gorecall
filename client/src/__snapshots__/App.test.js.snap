// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="app"
>
  <img
    alt="logo"
    className="app-logo"
    src="logo.svg"
  />
</div>
`;
